Tests/SentryTests/SentryCrash/SentryBinaryImageCacheTests.m

```diff
@@ -0,0 +1,247 @@
+}
+
+- (void)testStopCacheTwice
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+    sentrycrashbic_stopCache();
+    [self assertBinaryImageCacheLength:0];
+    sentrycrashbic_stopCache();
+    [self assertBinaryImageCacheLength:0];
+}
+
+- (void)testAddNewImage
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    addBinaryImage([mach_headers_test_cache[5] pointerValue], 0);
+    mach_headers_expect_array =
+        [mach_headers_test_cache subarrayWithRange:NSMakeRange(0, 6)].mutableCopy;
+    [self assertBinaryImageCacheLength:6];
+    [self assertCachedBinaryImages];
+
+    addBinaryImage([mach_headers_test_cache[6] pointerValue], 0);
+    mach_headers_expect_array =
+        [mach_headers_test_cache subarrayWithRange:NSMakeRange(0, 7)].mutableCopy;
+    [self assertBinaryImageCacheLength:7];
+    [self assertCachedBinaryImages];
+}
+
+- (void)testAddInvalidHeader
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    addBinaryImage(0, 0);
+    [self assertBinaryImageCacheLength:5];
+}
+
+- (void)testAddNewImageAfterStopping
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    sentrycrashbic_stopCache();
+    addBinaryImage([mach_headers_test_cache[6] pointerValue], 0);
+    [self assertBinaryImageCacheLength:0];
+}
+
+- (void)testRemoveImageFromTail
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    removeBinaryImage([mach_headers_expect_array[4] pointerValue], 0);
+    [self assertBinaryImageCacheLength:4];
+    [self assertCachedBinaryImages];
+
+    removeBinaryImage([mach_headers_expect_array[3] pointerValue], 0);
+    [self assertBinaryImageCacheLength:3];
+    [self assertCachedBinaryImages];
+}
+
+- (void)testRemoveImageFromBeginning
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    removeBinaryImage([mach_headers_expect_array[0] pointerValue], 0);
+    [self assertBinaryImageCacheLength:4];
+    [mach_headers_expect_array removeObjectAtIndex:0];
+    [self assertCachedBinaryImages];
+
+    removeBinaryImage([mach_headers_expect_array[0] pointerValue], 0);
+    [self assertBinaryImageCacheLength:3];
+    [mach_headers_expect_array removeObjectAtIndex:0];
+    [self assertCachedBinaryImages];
+}
+
+- (void)testRemoveImageAddAgain
+{
+    sentrycrashbic_startCache();
+    [self assertBinaryImageCacheLength:5];
+
+    removeBinaryImage([mach_headers_expect_array[0] pointerValue], 0);
+    [self assertBinaryImageCacheLength:4];
+
+    NSValue *removeItem = mach_headers_expect_array[0];
+    [mach_headers_expect_array removeObjectAtIndex:0];
+    [self assertCachedBinaryImages];
+
+    addBinaryImage(removeItem.pointerValue, 0);
+    [self assertBinaryImageCacheLength:5];
+    [mach_headers_expect_array insertObject:removeItem atIndex:4];
+    [self assertCachedBinaryImages];
+}
+
+- (void)testAddBinaryImageInParallel
```

Comment:
`m`: I'm not too worried about calling `addBinaryImage` from multiple threads, but rather stopping from another thread while still adding. Ideally, we would test adding, iterating, and stopping from different threads.